includes/gateways/stripe/includes/admin/admin-actions.php

```diff
@@ -32,27 +32,83 @@ function give_stripe_connect_save_options() {
 
 	// If we don't have values here, bounce.
 	if (
-		! isset( $get_vars['stripe_publishable_key'] )
-		|| ! isset( $get_vars['stripe_user_id'] )
-		|| ! isset( $get_vars['stripe_access_token'] )
-		|| ! isset( $get_vars['stripe_access_token_test'] )
-		|| ! isset( $get_vars['connected'] )
+		! isset( $get_vars['stripe_publishable_key'] ) ||
+		! isset( $get_vars['stripe_user_id'] ) ||
+		! isset( $get_vars['stripe_access_token'] ) ||
+		! isset( $get_vars['stripe_access_token_test'] ) ||
+		! isset( $get_vars['connected'] )
 	) {
 		return;
 	}
 
-	// Update keys.
-	give_update_option( 'give_stripe_connected', $get_vars['connected'] );
-	give_update_option( 'give_stripe_user_id', $get_vars['stripe_user_id'] );
-	give_update_option( 'live_secret_key', $get_vars['stripe_access_token'] );
-	give_update_option( 'test_secret_key', $get_vars['stripe_access_token_test'] );
-	give_update_option( 'live_publishable_key', $get_vars['stripe_publishable_key'] );
-	give_update_option( 'test_publishable_key', $get_vars['stripe_publishable_key_test'] );
+	// Unable to redirect, bail.
+	if ( headers_sent() ) {
+		return;
+	}
+
+	$stripe_account_id = $get_vars['stripe_user_id'];
+	$stripe_accounts   = give_stripe_get_all_accounts();
+	$all_account_slugs = array_keys( $stripe_accounts );
+	$accounts_count    = is_countable( $stripe_accounts ) ? count( $stripe_accounts ) + 1 : 1;
+	$account_slug      = give_stripe_get_unique_account_slug( $all_account_slugs, $accounts_count );
+	$secret_key        = ! give_is_test_mode() ? $get_vars['stripe_access_token'] : $get_vars['stripe_access_token_test'];
+	$account_name      = give_stripe_convert_slug_to_title( $account_slug );
+	$account_email     = '';
+	$account_country   = '';
+
+	// Set API Key to fetch account details.
+	\Stripe\Stripe::setApiKey( $secret_key );
+
+	// Get Account Details.
+	$account_details = give_stripe_get_account_details( $stripe_account_id );
+
+	// Setup Account Details for Connected Stripe Accounts.
+	if ( ! empty( $account_details->id ) && 'account' === $account_details->object ) {
+		$account_name    = $account_details->business_profile->name;
+		$account_slug    = $account_details->id;
+		$account_email   = $account_details->email;
+		$account_country = $account_details->country;
+	}
+
+	// If the same Stripe account is already connected, then don't save it.
+	if ( in_array( $stripe_account_id, wp_list_pluck( $stripe_accounts, 'give_stripe_user_id' ), true ) ) {
+		return;
+	}
 
-	// Delete option for user API key.
-	give_delete_option( 'stripe_user_api_keys' );
+	// Set the first account as default.
+	if ( ! $stripe_accounts ) {
 
+		// Set the connect details of Stripe account as default Stripe account.
+		give_update_option( 'give_stripe_connected', $get_vars['connected'] );
```

Comment:
Please review usage of these key and update code accordingly as we discussed: https://givewp.slack.com/archives/C0FAGC83C/p1590141922067000